### ISSUE TITLE
Same url,can't hit the memory cache When there are ImageViews with different size

### ISSUE THREAD

**Issue by 835127729** [bumptech/glide#1892]
I'm using Glide to load the same image(same url) into two different size ImageView.
For exmaple,one is (100dp * 100dp),other is (120dp * 120dp).
I add the OnClickListener each ImageView so Glide will load image after I click.
```
String url = "https://qianniu.alicdn.com/emoticon/1484531928510/icon144x144.png?type=1&suffix=png&width=144&height=144&fileId=06a54ab0f55d7475abb9575b005f92d9.png&packid=55";
imageView1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                RequestOptions requestOptions = new RequestOptions();
                requestOptions.diskCacheStrategy(DiskCacheStrategy.ALL);
                Glide.with(MainActivity.this).asBitmap().load(url).into(imageView);
            }
        });
imageView2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                RequestOptions requestOptions = new RequestOptions();
                requestOptions.diskCacheStrategy(DiskCacheStrategy.ALL);
                Glide.with(MainActivity.this).asBitmap().load(url).into(imageView);
            }
        });
```
I found imageView2 still load from **DISK_CACHE** even if I click imageView1 firstly.
Then I read the code of Glide,And I found **activeResources** didn't been hit beacuse imageView1 and imageView2 having **different size**.

Besides,**LruResourceCache** contains nothing so I cant hit the memory cache.
I found only when imageView release its bitmap,and this bitmap will been cached into LruResourceCache.

so my question is:
1、when did the cache will been put into the LruResourceCache?
2、if there are two different size ImageView and I hope they can load from memory cache or activeResources, what should I do?

**Comment by sjudd** [comment]
LruResourceCache is also size based. If you want to get a memory cache hit, all parts of your load, including the size, must be the same.

**Comment by TWiStErRob** [comment]
If all your images are like the one you linked (i.e. have hardcoded sizes), or you don't mind wasting memory, you can use `.override(144, 144)` to fix the size of the load, leading to a match when you load into different targets.

**Comment by bluepix** [comment]
I have a similar issue, but I'm requesting the same image with the same size.  I have some ImageViews in a Fragment to load Bitmaps into when i enter the Fragment. I found that when I quit and instantiate that Fragment for several times, Glide loaded the same bitmap either from activeResources or  from the DiskCache, but never from LruResourceCache. I noticed that a resource is put into memory cache only when it is released and the reference count (acquired) turns to 0. But when I simply quit the Fragment or Activity the ImageView is in, will release() be called by Glide for me? or i need to call it explicitly?
